autograder/submitty_autograding_shipper.py

```diff
@@ -948,6 +953,25 @@ def checkout_vcs_repo(config, my_file):
                 #    # and check out the right version
                 #    subprocess.call(['git', 'checkout', '-b', 'grade', what_version])
 
+                # copy the subfolder we want to the old checkout path and remove the rest
+                if tmp_checkout != "":
+                    os.chdir(tmp_checkout)
+                    if vcs_subdirectory[0] == '/':
+                        vcs_subdirectory = vcs_subdirectory[1:]
+                    files = os.listdir(os.path.join(checkout_path, vcs_subdirectory))
+                    if files:
+                        for f in files:
+                            shutil.move(
+                                        os.path.join(
+                                                     checkout_path,
+                                                     vcs_subdirectory,
+                                                     f),
+                                        tmp_checkout)
+                    else:
+                        subprocess.check_call('not_a_commadn')
```

Comment:
What is the purpose of this?

Reply:
This is a hack we used to get this feature working. I am working on a cleaner solution.

Also I fixed the `tmp_checkout` variable reference and updated the language to say externally hosted. After I finish the cleaner solution, I will update the documentation based for creating this type of gradeable.

Reply:
We have been working on unit test to help confirm our new process. These tests are in [8803](https://github.com/Submitty/Submitty/pull/8803). Once these are merged we can add test to show the new functionality is working without a hack.

Reply:
This 'hack' has been changed to a try/except case, and corresponding tests have been added to show the changes are working without the hack. 

Also, the gradeable now shows 'Externally Hosted' not 'Submitty Hosted'
![Screenshot from 2023-04-04 17-34-53](https://user-images.githubusercontent.com/46759635/229953433-239b6a54-647d-45ad-8bf6-bce5d4c1cd24.png)